auth-service/src/test/java/com/petclinic/authservice/businesslayer/AuthServiceUserServiceTests.java

```diff
@@ -202,7 +217,57 @@ public void testUpdateResetPasswordToken_UserNotFound() {
         // IllegalArgumentException should be thrown
     }
 
+    @Test
+    @DisplayName("updatePassword should throw NotFoundException for non-existing user")
+    void updatePassword_ShouldThrowNotFoundExceptionForNonExistingUser() {
+        // Arrange
+        String newPassword = "newPassword";
+        String token = "validToken";
+
+        ResetPasswordToken validToken = new ResetPasswordToken();
+        validToken.setUserIdentifier(1L);
+        validToken.setExpiryDate(new Date(System.currentTimeMillis() + 1000));
+
+        when(tokenRepository.findResetPasswordTokenByToken(any())).thenReturn(validToken);
+
+        when(userRepo.findById(validToken.getUserIdentifier())).thenReturn(Optional.empty());
+
+        // Act and Assert
+        NotFoundException exception = assertThrows(NotFoundException.class, () -> {
+            userService.updatePassword(newPassword, token);
+        });
+
+        assertEquals("Could not find any customer with the token " + token, exception.getMessage());
+
+        verify(userRepo).findById(validToken.getUserIdentifier());
+    }
+
 
+// Next Story
```

Comment:
What does this next story part mean?

Reply:
it means its a test that works but since some of the exceptions being used are wrong I'll do it in the next story concerning it.